schemas/manifest.schema.json

```
@@ -0,0 +1,473 @@
+                                        },
+                                        "key": {
+                                            "type": "string"
+                                        },
+                                        "packages": {"$ref": "#/$defs/string_or_array_of"}
+                                    }
+                                }
+                            }
+                        }
+                    }
+                },
+                "data_dir": {
+                    "type": "object",
+                    "required": [],
+                    "additionalProperties": false,
+                    "properties": {
+                        "dir": {
+                            "type": "string",
+                            "pattern": "^/.*$"
+                        },
+                        "subdirs": {
+                            "type": "array",
+                            "items": {
+                                "type": "string"
+                            }
+                        },
+                        "owner": {
+                            "type": "string",
+                            "pattern": "^[^:]*(:[rwx-]{3}$)?"
+                        },
+                        "group": {
+                            "type": "string",
+                            "pattern": "^[^:]*(:[rwx-]{3}$)?"
+                        }
+                    }
+                },
+                "database": {
+                    "type": "object",
+                    "required": ["type"],
+                    "additionalProperties": false,
+                    "properties": {
+                        "type": {
+                            "type": "string",
+                            "enum": ["mysql", "postgresql"]
+                        }
+                    }
+                },
+                "install_dir": {
+                    "type": "object",
+                    "required": [],
+                    "additionalProperties": false,
+                    "properties": {
+                        "dir": {
+                            "type": "string",
+                            "pattern": "^/.*$"
+                        },
+                        "owner": {
+                            "type": "string",
+                            "pattern": "^[^:]*(:[rwx-]{3}$)?"
+                        },
+                        "group": {
+                            "type": "string",
+                            "pattern": "^[^:]*(:[rwx-]{3}$)?"
+                        }
+                    }
+                },
+                "permissions": {
+                    "type": "object",
+                    "required": [],
+                    "additionalProperties": false,
+                    "patternProperties": {
+                        "^.[a-z][a-z0-9_]*$": {
+                            "type": "object",
+                            "required": ["url"],
```

Zblerg turns out `url` is not required because sometimes they are only internal / pure-LDAP ... though could also be symptomatic of a typo ...

Not sure what do, but I think I would remove it from the mandatory fields

```suggestion
                            "required": [],
```